packs/salt/actions/lib/utils.py

```diff
@@ -0,0 +1,30 @@
+import yaml
+
+action_meta = {
+    "name": "",
+    "parameters": {
+        "action": {
+            "type": "string",
+            "immutable": True,
+            "default": ""
+        },
+        "kwargs": {
+            "type": "object",
+            "required": False
+        }
+    },
+    "runner_type": "run-python",
+    "description": "Run Salt Runner functions through Salt API",
+    "enabled": True,
+    "entry_point": "runner.py"}
+
+
+def generate_action(module_type, action):
```

Comment:
No magic.  I run this function by hand given the yaml files have to be present for the action to be registered. I followed a working example [present in st2incubator](https://github.com/StackStorm/st2incubator/blob/master/etc/voodoo/etc/metagen.py#L131)